### DISCUSSION TITLE
Reading and Writing to Text files in a project and "\n" returns

### DISCUSSION THREAD

**Discussion by DauntlessK** [pyscript/pyscript#1924] [Q&A]
Forgive what may be a basic question or two...

• How can I read and write to text files within a project? It does not seem to work like having it running off pycharm locally. I've read some stuff on here but haven't gotten anywhere with it.

• I also noticed issues when the console prints text that contains "\n"... is there a workaround or substitution that will work in place of it?

Thanks for any help.

**Comment by WebReflection** [comment]
PyScript offers an `interpreter.writeFile` but basically it's the interpreter that exposes virtual FS (in memory) and you can use the interpreter FS to read or write anything, just like we do already to provide the stdlib.

The `"\n"` char AFAIK works just fine and it's used in both unit and integration tests ... what is the issue you are having there?
